### DISCUSSION TITLE
Ability to define allowable alert tag names and values

### DISCUSSION THREAD

**Discussion by raags** [grafana/grafana#41393] [Alerting]
**What would you like to be added**:
Enhance tags added to alerts in the Alert tab:
1. Ability to specify list of allowed tag names and values. This can be an autocomplete list that the user gets when adding a tag for an alert.
2. Admin page to allow viewing all the tags (Name and Values) added across all alerts.

**Why is this needed**:
Tags allow downstream alert aggregation and routing to work. But these can get inconsistent, due to typos, or someone forgetting to add an important tag. By making allow these to be defined by an administrator, as well as making them easy to use (via auto-complete / drop-down list), reduces errors, and ensure consistency across all alerts.

**Comment by jdmaloney** [comment]
Seconding this feature request.  Not sure what is already planned for the alerting revamp that's underway, but the ability to tag alerts and then take actions on them based on those tags (eg. pause all alerts with a given tag, start all alerts with a given tag, etc.) would be incredibly useful.  Right now we have some hacked together scripts that handle this functionality but it requires very precise alert naming which can be tedious.

**Comment by peterholmberg** [comment]
Converting this to a discussion since that's how we want to handle feature requests going forward.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
